fish-rust/src/wutil/format/format.rs

```diff
@@ -508,3 +508,9 @@ impl Printf for WString {
         self.as_utfstr().format(spec)
     }
 }
+
+impl Printf for &WString {
+    fn format(&self, spec: &ConversionSpecifier) -> Result<WString> {
+        self.as_utfstr().format(spec)
```

Comment:
untested but we can probably merge this with the implementation for `&wstr` by implementing for `AsRef<wstr>`

Reply:
That impl already exists upstream, and it wasn't enough: https://docs.rs/widestring/latest/widestring/utfstring/struct.Utf32String.html#impl-AsRef%3CUtf32Str%3E-for-Utf32String